lib/email/sender.rb

```diff
@@ -245,6 +251,13 @@ def send
       # can debug deliverability issues.
       email_log.smtp_group_id = smtp_group_id
 
+      # Store contents of all outgoing emails for greater visibility. Not
+      # really required by default, but a good idea to turn on for SMTP
+      # and IMAP sending.
+      if SiteSetting.enable_raw_outbound_email_logging
+        email_log.raw = Email::Cleaner.new(@message).execute
```

Comment:
I'm not sure if we should be storing this in the table because I don't think this site settings can be enabled for a long duration. On sites where alot of emails are sent, storing the raw in the database is going to make the `email_logs` table take up more disk space. Instead, could we just log the raw if a dev really needs to look at the raw message for debugging purposes?

Reply:
We log all of the IncomingEmail raw emails unconditionally, so I think this is a good compromise. Leaving it up to a dev enable it after the fact is no good...the horse has already bolted at that point, you can't log the raw of an email that has already been sent.

Now that I think about it though, I think this should just log unconditionally if the email is being sent via group SMTP. That way the impact will be minimal (only one or two groups will ever have this enabled on a site, and we are emailing multiple people with one EmailLog record now that we are going to be doing the CC addresses). Will change to do that and get rid of the setting.

Reply:
Also, we delete all EmailLog records after 90 days so that should lessen the impact even more.

Reply:
> Leaving it up to a dev enable it after the fact is no good...the horse has already bolted at that point, you can't log the raw of an email that has already been sent.

I understand the concern here but I guess this is a tradeoff we make. If problems with SMTP and IMAP sending rarely happens, we're using up more resources for little benefit. 

> I think this should just log unconditionally if the email is being sent via group SMTP. That way the impact will be minimal (only one or two groups will ever have this enabled on a site, and we are emailing multiple people with one EmailLog record now that we are going to be doing the CC addresses).

I guess if this isn't a widely used feature we can just log it in the DB when SMTP and IMAP is enabled but I'll keep an eye on the disk usage. The comparison to `IncomingEmail` isn't really fair because we send out way more emails than we receive.

Reply:
> DB when SMTP and IMAP is enabled

It's even less than that; it will only be if SMTP/IMAP is configured for a group. So far we've sent 403 emails to non-Group users (I am changing it so the outbound group SMTP only sends to topic allowed users), and out of those many will be compressed into one email with several CC addresses in a future change, so I think this logging is acceptable, especially when we are deleting the logs after 90 days. We can revisit if it really is taking up heaps of space in the DB but I doubt that.

We could also make it a group-level setting at some point that is default off if needed.

Reply:
Thank you for the clarification, lets just log in the DB as it is and keep an :eye: on the disk space usage.

Reply:
I changed it already to just log if group has SMTP enabled :) It is better than doing it the site setting way anyway, we don't need to log the full body of 98% of the emails we send